src/data_source/storage/sql/queries/availability.rs

```diff
@@ -96,6 +97,31 @@ where
         Ok(payload)
     }
 
+    async fn get_payload_metadata(
+        &mut self,
+        id: BlockId<Types>,
+    ) -> QueryResult<PayloadMetadata<Types>> {
+        let mut query = QueryBuilder::default();
+        let where_clause = query.header_where_clause(id)?;
+        // ORDER BY h.height ASC ensures that if there are duplicate blocks (this can happen when
+        // selecting by payload ID, as payloads are not unique), we return the first one.
+        let sql = format!(
+            "SELECT {PAYLOAD_METADATA_COLUMNS}
+              FROM header AS h
+              JOIN payload AS p ON h.height = p.height
+              WHERE {where_clause} AND p.num_transactions != NULL
```

Comment:
```suggestion
              WHERE {where_clause} AND p.num_transactions IS NOT NULL
```
I think you can not compare NULL with equality operators in postgres